audinterface/core/process.py

```diff
@@ -806,22 +809,30 @@ def process_signal_from_index(
         Raises:
             RuntimeError: if sampling rates do not match
             RuntimeError: if channel selection is invalid
+            RuntimeError: if sequence returned by ``process_func``
+                does not match length of ``index``
             ValueError: if index contains duplicates
 
         .. _audformat: https://audeering.github.io/audformat/data-format.html
 
         """
         utils.assert_index(index)
 
-        starts_i, ends_i = utils.segments_to_indices(
-            signal, sampling_rate, index,
-        )
-        y = self(signal, sampling_rate, starts_i, ends_i)
-
-        # For an empty Series we force the dtype
-        if len(y) == 0:
-            y = pd.Series(y, index=index, dtype=object)
+        if len(index) == 0:
+            y = pd.Series([], index=index, dtype=object)
         else:
+            starts_i, ends_i = utils.segments_to_indices(
+                signal, sampling_rate, index,
+            )
+            y = self(signal, sampling_rate, starts_i, ends_i)
+            if (
+                    not isinstance(y, collections.abc.Iterable)
```

Comment:
does this capture the empty case or what is it for?

Reply:
The first one capture the case of a single returned value, for which we get an error when doing `len(y)`

Reply:
According to the type hints the function must always return a sequence:

![image](https://user-images.githubusercontent.com/10383417/175492798-8053604b-e842-459a-97a3-32f1d9a38d72.png)

So theoretically we don't have to check for this case :)

Reply:
But we want to raise our custom error message here, this is the reason why we check for it as well.

Reply:
I mean in principle you are right, but as we raise already the custom error, I thought we could also include the case where no sequence is provided.

Reply:
Yes, but as long as the user sticks to the type hints, you can safely apply `len(y)`. But ok, doesn't matter.